controllers/storagecluster/cephcluster.go

```diff
@@ -110,6 +130,37 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		return reconcile.Result{}, fmt.Errorf("'StorageDeviceSets' should not be initialized in an external CephCluster")
 	}
 
+	// remove the fix in the next release 4.19
+	// as it is needed only for the upgrade to 4.18
+	if !sc.Spec.ExternalStorage.Enable {
+		// delete the osd prepare jobs, which are completed and have duplicated placements
+		jobName := "rook-ceph-osd-prepare"
+		listOpts := &client.ListOptions{
+			Namespace: sc.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				"app": jobName,
+			}),
+		}
+		jobs := &batchv1.JobList{}
+		// list all the prepare jobs
+		err := r.Client.List(r.ctx, jobs, listOpts)
+		if err != nil {
```

Comment:
You need to ignore Notfound errors or else the reconciler will always return from here during cluter creation or when the duplicated jobs are already deleted.

Reply:
I am not very sure if client.List returns a error or empty list if it doesn't find anything, so please correct me if I am wrong.

Reply:
We are listing the job, so in case there is no job, it will return empty slice